src/Editor.js

```diff
@@ -514,8 +515,36 @@ const Editor = () => {
                   true
                 );
               }}
+              onDoubleClick={() => setEditableListItemId(id)}
+              onBlur={() => setEditableListItemId("")}
             >
-              {name}
+              {editableListItemId === id ? (
+                <input
+                  type="text"
+                  autoFocus
+                  onChange={event => {
+                    event.persist();
+                    const updatedName = event.currentTarget.value;
+                    setState(current => ({
+                      ...current,
+                      doc: {
+                        ...current.doc,
+                        layers: current.doc.layers.map(layer => {
+                          if (layer.id === id) {
+                            const updatedLayer = layer;
+                            updatedLayer.name = updatedName;
+                            return updatedLayer;
+                          }
+                          return layer;
```

Comment:
We're mutating state here because `updatedLayer` === `layer` (make changes to `updatedLayer` is the same as making changes to `layer` because they both point to the same object in memory). To avoid mutations, we can instead write it like this:

```js
return layer.id === id ? { ...layer, name: updatedName } : layer
```